DiceKit/FrequencyDistributionIndex.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  FrequencyDistributionIndex.swift
+//  DiceKit
+//
+//  Created by Brentley Jones on 8/1/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+public struct FrequencyDistributionIndex: ForwardIndexType {
+    
+    typealias OrderedValues = [FrequencyDistribution.Outcome]
+    
+    let index: Int
+    let orderedOutcomes: OrderedValues
+    
+    static func startIndex(orderedOutcomes: OrderedValues) -> FrequencyDistributionIndex {
+        return FrequencyDistributionIndex(index: 0, orderedOutcomes: orderedOutcomes)
+    }
+    
+    static func endIndex(orderedOutcomes: OrderedValues) -> FrequencyDistributionIndex {
+        return FrequencyDistributionIndex(index: orderedOutcomes.count, orderedOutcomes: orderedOutcomes)
```

Comment:
No. Look at the `Indexable` protocol. `endIndex` is "one past the end". It's used as an "EOF" indicator.